javav2/usecases/topics_and_queues/src/main/java/com/example/sns/SNSWorkflow.java

```diff
@@ -0,0 +1,669 @@
+
+        System.out.println(DASHES);
+        System.out.println("9. Delete the received message. Press any key to continue.");
+        in.nextLine();
+        deleteMessages(sqsClient, sqsQueueUrl, messageList);
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("10. Unsubscribe from the topic and delete the queue. Press any key to continue.");
+        in.nextLine();
+        unSub(snsClient, subscriptionArn);
+        deleteSQSQueue(sqsClient, sqsQueueName);
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("11. Delete the topic. Press any key to continue.");
+        in.nextLine();
+        deleteSNSTopic(snsClient, topicArn);
+
+        System.out.println(DASHES);
+        System.out.println("The SNS/SQS workflow has completed successfully.");
+        System.out.println(DASHES);
+    }
+
+    public static void deleteSNSTopic(SnsClient snsClient, String topicArn ) {
+        try {
+            DeleteTopicRequest request = DeleteTopicRequest.builder()
+                .topicArn(topicArn)
+                .build();
+
+            DeleteTopicResponse result = snsClient.deleteTopic(request);
+            System.out.println("Status was " + result.sdkHttpResponse().statusCode());
+
+        } catch (SnsException e) {
+            System.err.println(e.awsErrorDetails().errorMessage());
+            System.exit(1);
+        }
+    }
+
+    public static void deleteSQSQueue(SqsClient sqsClient, String queueName) {
+        try {
+            GetQueueUrlRequest getQueueRequest = GetQueueUrlRequest.builder()
+                .queueName(queueName)
+                .build();
+
+            String queueUrl = sqsClient.getQueueUrl(getQueueRequest).queueUrl();
+            DeleteQueueRequest deleteQueueRequest = DeleteQueueRequest.builder()
+                .queueUrl(queueUrl)
+                .build();
+
+            sqsClient.deleteQueue(deleteQueueRequest);
+            System.out.println(queueName +" was successfully deleted.");
+
+        } catch (SqsException e) {
+            System.err.println(e.awsErrorDetails().errorMessage());
+            System.exit(1);
+        }
+    }
+
+    public static void unSub(SnsClient snsClient, String subscriptionArn) {
+        try {
+            UnsubscribeRequest request = UnsubscribeRequest.builder()
+                .subscriptionArn(subscriptionArn)
+                .build();
+
+            UnsubscribeResponse result = snsClient.unsubscribe(request);
+            System.out.println("Status was " + result.sdkHttpResponse().statusCode()
+                + "\nSubscription was removed for " + request.subscriptionArn());
+
+        } catch (SnsException e) {
+            System.err.println(e.awsErrorDetails().errorMessage());
+            System.exit(1);
+        }
+    }
+
+    public static void deleteMessages(SqsClient sqsClient, String queueUrl, List<Message> messages) {
```

Comment:
I think this needs an SOS addition, "sqs_DeleteMessageBatch"

Reply:
I will add that snippet to this code.